src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceServiceTest.kt

```diff
@@ -2368,6 +2437,81 @@ class LicenceServiceTest {
     )
   }
 
+  @Test
+  fun `approving an HDC licence variation`() {
+    val variation = createHdcVariationLicence().copy(id = 2, variationOfId = 1L)
+    whenever(licenceRepository.findById(2L)).thenReturn(Optional.of(variation))
+    whenever(staffRepository.findByUsernameIgnoreCase("smills")).thenReturn(aCom)
+
+    val licenceCaptor = ArgumentCaptor.forClass(EntityLicence::class.java)
+    val eventCaptor = ArgumentCaptor.forClass(EntityLicenceEvent::class.java)
+    val auditCaptor = ArgumentCaptor.forClass(EntityAuditEvent::class.java)
+
+    service.approveLicenceVariation(2L)
+
+    verify(licenceRepository, times(1)).findById(2L)
+
+    // Capture calls to licence, history and audit saveAndFlush - as a list
+    verify(licenceRepository, times(1)).saveAndFlush(licenceCaptor.capture())
+    verify(licenceEventRepository, times(1)).saveAndFlush(eventCaptor.capture())
+    verify(auditEventRepository, times(1)).saveAndFlush(auditCaptor.capture())
+    verify(staffRepository, times(1)).findByUsernameIgnoreCase(aCom.username)
+
+    // Check all calls were made
+    assertThat(licenceCaptor.allValues.size).isEqualTo(1)
+    assertThat(eventCaptor.allValues.size).isEqualTo(1)
+    assertThat(auditCaptor.allValues.size).isEqualTo(1)
+
+    assertThat(licenceCaptor.allValues[0])
+      .extracting("id", "statusCode", "updatedByUsername", "approvedByUsername", "approvedByName", "updatedBy")
+      .isEqualTo(listOf(2L, LicenceStatus.VARIATION_APPROVED, aCom.username, aCom.username, "X Y", aCom))
```

Comment:
question: do we want `LicenceStatus` and `LicenceEventType` s for HDC variations to be prefaced with `HDC_`? For example the preexisting `HDC_VARIATION_CREATED` and `HDC_VARIATION_CREATED`